SU2_CFD/src/solvers/CTurbSSTSolver.cpp

```diff
@@ -1034,10 +1039,29 @@ su2double CTurbSSTSolver::GetInletAtVertex(su2double *val_inlet,
 
 void CTurbSSTSolver::SetUniformInlet(const CConfig* config, unsigned short iMarker) {
   if (config->GetMarker_All_KindBC(iMarker) == INLET_FLOW) {
+    const string Marker_Tag = config->GetMarker_All_TagBound(iMarker);
     for (unsigned long iVertex = 0; iVertex < nVertex[iMarker]; iVertex++) {
-      Inlet_TurbVars[iMarker][iVertex][0] = GetTke_Inf();
-      Inlet_TurbVars[iMarker][iVertex][1] = GetOmega_Inf();
+      if (config->GetKind_Species_Model() != SPECIES_MODEL::NONE) {
+        CFluidModel* FluidModel= nullptr;
+        FluidModel = new CFluidScalar(config->GetSpecific_Heat_Cp(), config->GetGas_Constant(), config->GetPressure_Thermodynamic(), config);
+        FluidModel->SetTDState_T(config->GetInlet_Ttotal(Marker_Tag), config->GetInlet_SpeciesVal(Marker_Tag));
+        const su2double VelMag2_inlet = pow(config->GetInlet_Ptotal(Marker_Tag),2);
```

Comment:
What you had before was good... Now this is wrong for everything except velocity inlets